### PR TITLE
CFE-3198: Added between_versions macro to parser

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4549/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4549/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4549/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4552/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4552/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4552/

File: libpromises/cf3lex.l [hunk 1]
```diff
@@ -217,6 +218,45 @@ promise_type   [a-zA-Z_]+:
                         }
                       }
 
+{macro_if_between_versions} {
+                        if ( P.line_pos != 1 )
```
Comment (karlhto) [inline libpromises/cf3lex.l]: Why inconsistent spacing with the rest of the function?

File: libpromises/cf3lex.l [hunk 2]
```diff
@@ -88,9 +88,10 @@ eat_line  ([\n]|[\xd][\xa]).*
 
 comment    #[^\n]*
 
-macro_if_minimum_version @if\ minimum_version\([0-9]{1,10}\.[0-9]{1,10}(\.[0-9]{1,10})?\)
-macro_if_maximum_version @if\ maximum_version\([0-9]{1,10}\.[0-9]{1,10}(\.[0-9]{1,10})?\)
-macro_if_feature         @if\ feature\([a-zA-Z0-9_]+\)
+macro_if_minimum_version  @if\ minimum_version\([0-9]{1,5}(\.[0-9]{1,5}){0,2}\)
```
Comment (karlhto) [inline libpromises/cf3lex.l]: Nice refactor : )

File: libpromises/cf3lex.l (hunk 1 above)
Comment (olehermanse) [reply libpromises/cf3lex.l]: I hate this indentation style. I want to fix it later. With longer rule names, the code just goes off the side of the editor.

**Comment by olehermanse** [comment]
> I feel like between implies exclusivity, but I don't have a good suggestion for anything else.

@karlhto I agree about the word itself, but when using it, I don't think it's a big problem, it is intuitive to me that it is inclusive:

```
@if between_versions(3.15.0, 3.15.5)
```

I think in most cases it makes more sense that it is inclusive. (The name is taken from ticket, by the way.)

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4556/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4556/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4556/

**Pull request merged**